analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -145,4 +146,7 @@ public override void ExecutionCompleted()
             ReportIssue(operation, operation.Syntax.ToString());
         }
     }
+
+    private static bool IsEmptyArray(IOperation operation) =>
+        operation.AsArrayCreation()?.DimensionSizes.Any(x => x.AsLiteral() is { WrappedOperation.ConstantValue.Value: 0 }) ?? false;
```

Comment:
Please add a test cases with `const`. That should break this. And it should get fixed by 
```suggestion
        operation.AsArrayCreation()?.DimensionSizes.Any(x => x.ConstantValue.Value is 0) ?? false;
```

Reply:
It should break it. Add Noncompliant test case when the const is 0.

Also this should break it:
```
var array = new int[2-2];
array.Clone();                  // Noncompliant
```